pkg/highlight/highlighter.go

```diff
@@ -466,7 +467,9 @@ func (h *Highlighter) Highlight(input LineStates, startline, endline int) {
 		input.Unlock()
 	}
 
-	for i := startline; i <= endline; i++ {
+	for i := startline; ; i++ {
+		var lastState *region
```

Comment:
nit: it is a bit more Go-idiomatic to declare a variable where it is first used (i.e. before `if i >= endline` in this case).